target/linux/generic/hack-6.6/736-net-phy-realtek-add-rtl8251l-rtl8254b-rtl8261be-rtl8261n.patch

```diff
@@ -0,0 +1,713 @@
++				     const struct phy_driver *phydrv)
++{
++	if (!rtlgen_is_c45_match(phydev, PHYID_RTL8261N, true))
++		return 0;
++
++	if (!phydev->mdio.bus->read_c45)
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
++	if (data < 0)
++		return 0;
++
++	if (data == 0x8251)
++		return 0;
++
++	data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x104);
++	if (data < 0)
++		return 0;
++
++	if ((data & 0xFFC0) == 0x1140)
++		return 0;
++
++	return 1;
++}
++
++static int rtl8264b_match_phy_device(struct phy_device *phydev,
++				     const struct phy_driver *phydrv)
++{
++	if (!rtlgen_is_c45_match(phydev, PHYID_RTL8264B, true))
++		return 0;
++
++	if (!phydev->mdio.bus->read_c45)
++		return 0;
++
++	int data = phy_read_mmd(phydev, MDIO_MMD_VEND1, 0x103);
++	if (data < 0)
++		return 0;
++
++	if (data == 0x8254)
++		return 0;
++
++	return 1;
++}
++
+ static int rtlgen_resume(struct phy_device *phydev)
+ {
+ 	int ret = genphy_resume(phydev);
+@@ -1705,6 +1822,394 @@ static irqreturn_t rtl8221b_handle_interrupt(struct phy_device *phydev)
+ 	return IRQ_HANDLED;
+ }
+ 
++static int rtl826x_probe(struct phy_device *phydev)
++{
++	struct device *dev = &phydev->mdio.dev;
++	struct rtl826x_priv *priv = NULL;
++
++	priv = devm_kzalloc(dev, sizeof(struct rtl826x_priv), GFP_KERNEL);
++	if (!priv)
++		return -ENOMEM;
++
++	memset(priv, 0, sizeof(struct rtl826x_priv));
++
++	priv->pnswap_tx = device_property_read_bool(dev, "realtek,pnswap-tx");
++	priv->pnswap_rx = device_property_read_bool(dev, "realtek,pnswap-rx");
++	priv->patch_rtk_serdes = device_property_read_bool(dev, "realtek,patch-rtk-serdes");
++	phydev->priv = priv;
++
++	return 0;
++}
++
++static int rtl8261n_probe(struct phy_device *phydev)
++{
++	struct rtl826x_priv *priv;
++	int ret;
++
++	ret = rtl826x_probe(phydev);
++	if (ret < 0)
++		return ret;
++
++	priv = phydev->priv;
++
++	if (phy_patch_rtl8261n_db_init(phydev, &priv->patch) < 0)
++		return -ENOMEM;
++
++	return 0;
++}
++
++static int rtl8264b_probe(struct phy_device *phydev)
++{
++	struct rtl826x_priv *priv;
++	int ret;
++
++	ret = rtl826x_probe(phydev);
++	if (ret < 0)
++		return ret;
++
++	priv = phydev->priv;
++
++	if (phy_patch_rtl8264b_db_init(phydev, &priv->patch) < 0)
++		return -ENOMEM;
++
++	return 0;
++}
++
++static int rtl826x_config_init(struct phy_device *phydev)
++{
++	struct rtl826x_priv *priv = phydev->priv;
++	int ret = 0;
++
++	/* toggle reset */
++	phy_modify_mmd_changed(phydev, MDIO_MMD_VEND1, 0x145, BIT(0), 1);
++	phy_modify_mmd_changed(phydev, MDIO_MMD_VEND1, 0x145, BIT(0), 0);
++	mdelay(30);
++
++	ret = phy_patch(phydev);
++	if (ret) {
++		phydev_err(phydev, "patch failed!! 0x%X\n", ret);
++		return ret;
++	}
++
++	if (priv->pnswap_tx)
++		phy_set_bits_mmd(phydev, MDIO_MMD_VEND1, RTL8261N_SERDES_GLOBAL_CFG, RTL8261N_SERDES_GLOBAL_CFG_HSO_INV);
++	if (priv->pnswap_rx)
++		phy_set_bits_mmd(phydev, MDIO_MMD_VEND1, RTL8261N_SERDES_GLOBAL_CFG, RTL8261N_SERDES_GLOBAL_CFG_HSI_INV);
```

Comment:
Going to ask if this bit was actually default set again, given it's possible we may have been looking at the wrong register last time I asked.

Reply:
I checked 0xc1 since too. by default it had 0x215 in it. so i think both bits are not set by default.